pandas/core/algorithms.py

```diff
@@ -65,6 +65,8 @@ def _ensure_data(values, dtype=None):
 
     # we check some simple dtypes first
     try:
+        if is_object_dtype(dtype):
```

Comment:
did you run the asv? this can degrade as its in a critical path. notice there is *already* a check for object type later on. This function  receives LOTS of input.